packages/js/product-editor/src/components/block-editor/block-editor.tsx

```diff
@@ -119,18 +121,20 @@ export function BlockEditor( {
 		};
 	}, [ canUserCreateMedia, _settings ] );
 
-	const [ productType ] = useProductEntityProp< Product[ 'type' ] >( 'type', {
-		postType,
-	} );
-
 	const [ productTemplateId ] = useProductEntityProp< string >(
 		'meta_data._product_template_id',
 		{ postType }
 	);
 
+	const { record: product, isResolving } = useEntityRecord(
+		'postType',
+		postType,
+		productId
+	);
+
 	const { productTemplate } = useProductTemplate(
 		productTemplateId,
-		productType
+		! isResolving ? product || null : undefined
```

Comment:
If the product is not resolving, does it return `undefined` if the product doesn't exist? Just wondering why the `product || null`.

Reply:
I've checked the newest type definitions and debugged: It returns `undefined` while still resolving, and `null` if it fails to resolve. So in my opinion, we don't need to check `isResolving`, just handle the `null` in the signature.

Also: I wonder if we could update some package versions in the product-editor, like `@wordpress/core-data`, since we are already using the new modules anyway. It has type definitions in the newest version.